operator/v1/types_ingress.go

```diff
@@ -176,6 +176,17 @@ type IngressControllerSpec struct {
 	//
 	// +optional
 	HTTPHeaders *IngressControllerHTTPHeaders `json:"httpHeaders,omitempty"`
+
+	// httpHeaderBuffer defines parameters for header buffer size values.
+	// If this field is empty, the default values set within the
+	// IngressController will be used. Setting this field is generally not
+	// recommended as header buffer values that are too small may break the
+	// IngressController and header buffer values that are too large could
+	// cause the IngressController to use significantly more compute resources
+	// than necessary.
+	//
+	// +optional
+	HTTPHeaderBuffer *IngressControllerHTTPHeaderBuffer `json:"httpHeaderBuffer,omitempty"`
```

Comment:
We try not to introduce pointers in API fields unnecessarily.  Can we make this field hold a non-pointer value, and treat zero values as unspecified (since each subfield has a positive minimum value)?  

One implication of the above suggestion is that this field's subfields would be defaulted independently (i.e., `spec.httpHeaderBuffer.headerBufferSize` could be specified and `spec.httpHeaderBuffer.headerBufferMaxRewrite` omitted or vice versa, the specified value would be used, and the omitted value would be defaulted).  Is there any problem with defaulting those two fields independently of each other?

Reply:
> Is there any problem with defaulting those two fields independently of each other?
I don't think there would be any problem with doing this. I'll update the PR accordingly.

One question though, how can one of these subfields be omitted if they both have the `+required` tag?

Reply:
Good question.  Looking at the generated yaml, it looks like the `+required` tag isn't having an effect.  I think you need to add `// +kubebuilder:validation:Required` as well.  Once you do that, I believe the generated yaml will have `httpHeaderBuffer` optional and its subfields required.  

For a point of reference, I see that the auth config has optional status: https://github.com/openshift/api/blob/f67d7a61e06075e9c84bc731fca470fff2832227/config/v1/types_authentication.go#L19-L21
`AuthenticationStatus` has optional `integratedOAuthMetadata` (implied by the absence of `// +kubebuilder:validation:Required`): https://github.com/openshift/api/blob/f67d7a61e06075e9c84bc731fca470fff2832227/config/v1/types_authentication.go#L69-L83
`ConfigMapNameReference` has required `name`: https://github.com/openshift/api/blob/f67d7a61e06075e9c84bc731fca470fff2832227/config/v1/types.go#L18-L23
And the generated yaml reflects that `status` and `status.integratedOAuthMetadata` are optional but `status.integratedOAuthMetadata.name` is required: https://github.com/openshift/api/blob/f67d7a61e06075e9c84bc731fca470fff2832227/config/v1/0000_10_config-operator_01_authentication.crd.yaml#L136-L161

Reply:
I see, thanks for the in-depth explanation!

I was using the `// +kubebuilder:validation:required` instead of `// +kubebuilder:validation:Required`. * facepalm *

Fixed!